cdap-ui/app/cdap/components/PipelineConfigurations/ConfigurationsContent/PipelineConfig.js

```diff
@@ -0,0 +1,260 @@
+)(Instrumentation);
+
+const mapStateToStageLoggingProps = (state) => {
+  return {
+    stageLogging: state.stageLoggingEnabled
+  };
+};
+const mapDispatchToStageLoggingProps = (dispatch) => {
+  return {
+    onToggle: (value) => {
+      dispatch({
+        type: PipelineConfigurationsActions.SET_STAGE_LOGGING,
+        payload: { stageLogging: value }
+      });
+    }
+  };
+};
+
+const StageLogging = ({stageLogging, onToggle}) => {
+  return (
+    <div className="label-with-toggle stageLogging row">
+      <span className="toggle-label col-xs-4">Stage Level Logging</span>
+      <div className="col-xs-7 toggle-container">
+        <ToggleSwitch
+          isOn={stageLogging}
+          onToggle={onToggle.bind(null, !stageLogging)}
+        />
+        <IconSVG
+          name="icon-info-circle"
+          id="stagelogging-info-icon"
+        />
+        <UncontrolledTooltip
+          target="stagelogging-info-icon"
+          delay={{show: 250, hide: 0}}
+          placement="right"
+        >
+          Allows logs from each stage in the pipeline to be queried individually. It is recommended to always have this setting on, unless the environment is short on resources.
+        </UncontrolledTooltip>
+      </div>
+    </div>
+  );
+};
+
+StageLogging.propTypes = {
+  stageLogging: PropTypes.bool,
+  onToggle: PropTypes.func
+};
+
+const ConnectedStageLogging = connect(
+  mapStateToStageLoggingProps,
+  mapDispatchToStageLoggingProps
+)(StageLogging);
+
+const mapStateToCheckpointingProps = (state) => {
+  return {
+    checkpointing: state.disableCheckpoints
+  };
+};
+const mapDispatchToCheckpointingProps = (dispatch) => {
+  return {
+    onToggle: (value) => {
+      dispatch({
+        type: PipelineConfigurationsActions.SET_CHECKPOINTING,
+        payload: { checkpointing: value }
+      });
+    }
+  };
+};
+
+const Checkpointing = ({checkpointing, onToggle}) => {
+  return (
+    <div className="label-with-toggle checkpointing row">
+      <span className="toggle-label col-xs-4">Checkpointing</span>
+      <div className="col-xs-7 toggle-container">
+        <ToggleSwitch
+          isOn={checkpointing}
+          onToggle={onToggle.bind(null, !checkpointing)}
+        />
+        <IconSVG
+          name="icon-info-circle"
+          id="checkpointing-info-icon"
+        />
+        <UncontrolledTooltip
+          target="checkpointing-info-icon"
+          delay={{show: 250, hide: 0}}
+          placement="right"
+        >
+          Allows Apache Spark Streaming to checkpoint data (RDDs) to persistent storage so that the pipeline can recover from failures.
+        </UncontrolledTooltip>
+      </div>
+    </div>
+  );
+};
+
+Checkpointing.propTypes = {
+  checkpointing: PropTypes.bool,
+  onToggle: PropTypes.func
+};
+
+const ConnectedCheckpointing = connect(
+  mapStateToCheckpointingProps,
+  mapDispatchToCheckpointingProps
+)(Checkpointing);
+
+const mapStateToSelectBatchIntervalRangeProps = (state) => {
+  return {
```

Comment:
Seems like they're not required, the classes still show up without the ownProps

Reply:
I think react passes a whitelist of props down to children because they are relevant for any react component I guess. If we need to pass in any custom prop then I guess we need to use `ownProps`.